File: veterinaria-system/frontend/src/models/Tutor.js
/**
 * Modelo Tutor
 * Arquitectura MVC - Capa de Modelos
 */

class Tutor {
  constructor(data = {}) {
    this.id = data.id || null;
    this.usuario = data.usuario || null;
    this.ci = data.ci || '';
    this.direccion = data.direccion || '';
    this.fecha_nacimiento = data.fecha_nacimiento || '';
    this.edad = data.edad || null;
    this.nombre_completo = data.nombre_completo || '';
    this.email = data.email || '';
    this.telefono = data.telefono || '';
    this.created_at = data.created_at || null;
    this.updated_at = data.updated_at || null;
  }

  static fromJSON(json) {
    return new Tutor(json);
  }

  toJSON() {
    return {
      id: this.id,
      usuario_id: this.usuario,
      ci: this.ci,
      direccion: this.direccion,
      fecha_nacimiento: this.fecha_nacimiento
    };
  }
}

export default Tutor;

File: veterinaria-system/backend/api/models/tutor.py
"""
Modelo Tutor - Tutores (dueños) de mascotas.
Arquitectura MVC - Capa de Modelos
"""
from django.db import models
from datetime import date
from .usuario import Usuario


class Tutor(models.Model):
    """
    Modelo para los tutores (dueños) de mascotas.
    Tiene relación ONE-TO-ONE con Usuario.
    """
    usuario = models.OneToOneField(Usuario, on_delete=models.CASCADE, related_name='tutor')
    ci = models.CharField(max_length=20, unique=True, verbose_name='Cédula de Identidad')
    direccion = models.TextField()
    fecha_nacimiento = models.DateField()
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        verbose_name = 'Tutor'
        verbose_name_plural = 'Tutores'
        ordering = ['-created_at']

    def __str__(self):
        return f"{self.usuario.nombre_completo} - CI: {self.ci}"

    @property
    def edad(self):
        """Calcula la edad del tutor"""
        today = date.today()
        return today.year - self.fecha_nacimiento.year - (
            (today.month, today.day) < (self.fecha_nacimiento.month, self.fecha_nacimiento.day)
        )
